interface/modules/zend_modules/module/Carecoordination/src/Carecoordination/Model/EncounterccdadispatchTable.php

```diff
@@ -27,6 +27,199 @@ public function __construct()
+                    $prow = $pres->current();
+                }
+            }
+
+            if ($okE == 1 || $okN == 2 || $okN == 1 || $okO == 1) {
+                $pid = $prow['pid'];
+                $GLOBALS['pid'] = $prow['pid'];
+            } elseif ($okE == 2 || $okN == 3) {
+                $pres = $appTable->zQuery("
+                    SELECT
+                        ad.audit_master_id
+                    FROM
+                        audit_details ad
+                        JOIN audit_details ad2
+                            ON ad2.audit_master_id = ad.audit_master_id
+                            AND ad2.table_name = 'patient_access_offsite'
+                            AND ad2.field_name = 'portal_pwd'
+                    WHERE ad.table_name = 'patient_access_offsite'
+                        AND ad.field_name = 'portal_username'
+                        AND ad.field_value = ?
+                ", array($credentials[6]));
+                $arow = $pres->current();
+                $auditmasterid = $arow['audit_master_id'];
+                $GLOBALS['auditmasterid'] = $arow['audit_master_id'];
+                $pid = 0;
+                $GLOBALS['pid'] = 0;
+            }
+
+            $_GET['site'] = $credentials[0];
+            if ($okE) {
+                if ($okE == 1) {
+                    $pres = $appTable->zQuery("SELECT allow_patient_portal FROM patient_data WHERE pid=?", array($pid));
+                    $portal = $pres->current();
+                } elseif ($okE == 2) {
+                    $pres = $appTable->zQuery("
+                        SELECT
+                            IF(COUNT(*),'yes','no') AS allow_patient_portal
+                        FROM
+                            audit_master am
+                            JOIN audit_details ad
+                                ON ad.audit_master_id = am.id
+                                AND ad.table_name = 'patient_access_offsite'
+                                AND ad.field_name = 'portal_username'
+                        WHERE am.approval_status = 1
+                            AND ad.field_value = ?
+                    ", array($credentials[6]));
+                    $portal = $pres->current();
+                }
+
+                if (strtolower($portal['allow_patient_portal']) != 'yes') {
+                    return false;
+                }
+            }
+
+            $GLOBALS['validated_offsite_portal'] = true;
+            if ($okO) {
+                return 'oemruser';
+            } elseif ($okE == 1) {
+                return 'existingpatient';
+            } elseif ($okE == 2) {
+                return 'newpatienttoapprove';
+            } elseif ($okN == 1 || $okN == 2) {
+                return 'newpatient';
+            } elseif ($okN == 3) {
+                return 'newpatienttoapprove';
+            }
+
+            return false;
+        } else {
+            return false;
+        }
+    }
+
```

Comment:
This block of code shouldn't be here (I removed this awhile back)

Reply:
Which line #'s? This was probably due to the countless times I had to rebase this

Reply:
entire block of code from line 30-222